position-node/src/workflows.test.ts

```diff
@@ -18,81 +39,57 @@ beforeAll(async () => {
       stdio: 'inherit',
     },
   });
-});
-
-afterAll(async () => {
-  await testEnv?.teardown();
-});
 
-function gameInfo(overrides: Partial<GameInfo> = {}): GameInfo {
-  const defaults = {
-    anchorX: 0,
-    anchorY: 0,
-    ballX: 0,
-    ballY: 0,
-    length: 0,
-    width: 0,
-    height: 0,
-    angle: 0,
-    angleAccel: 0,
-    angleVelocity: 0,
-    dt: 0,
-    speed: 0,
-  };
-  return { ...defaults, ...overrides };
-}
+  const { nativeConnection } = testEnv;
+  worker = await Worker.create({
+    connection: nativeConnection,
+    taskQueue,
+    workflowsPath: require.resolve('./workflows'),
+    activities: undefined,
+  });
 
-test('gameInfo without overrides', () => {
-  expect(gameInfo().anchorX).toBe(0);
+  runPromise = worker.run();
 });
 
-test('gameInfo with overrides', () => {
-  expect(gameInfo({ anchorX: 1 }).anchorX).toBe(1);
+beforeEach(async () => {
+  handle = await testEnv.workflowClient.start(pendulum, {
+    args: [{ ...gameInfo, anchorX: 1 }],
+    workflowId: uuid4(),
+    taskQueue,
+  });
 });
 
-type PendulumHandle = WorkflowHandleWithFirstExecutionRunId<(info: GameInfo) => Promise<unknown>>;
-type PendulumTest = (handle: PendulumHandle) => Promise<unknown>;
-
-async function runPendulum(initialInfo = gameInfo(), fn: PendulumTest = async () => undefined) {
-  const { workflowClient, nativeConnection } = testEnv;
-  const worker = await Worker.create({
-    connection: nativeConnection,
-    taskQueue: 'test',
-    workflowsPath: require.resolve('./workflows'),
-    activities: undefined,
-  });
-  await worker.runUntil(async () => {
-    const handle = await workflowClient.start(pendulum, {
-      args: [initialInfo],
-      workflowId: uuid4(),
-      taskQueue: 'test',
-    });
-    try {
-      await fn(handle); // invoke test
-    } finally {
-      await handle.signal(exitSignal);
-      await handle.result();
+afterEach(async() => {
+  await handle.terminate().catch(err => {
+    if (err instanceof WorkflowNotFoundError) {
+      return;
     }
+
+    throw err;
   });
-}
+});
+
+afterAll(async () => {
+  worker.shutdown();
+  await runPromise;
+
+  await testEnv?.teardown();
+});
 
 test('pendulum exitSignal', async () => {
-  await runPendulum();
+  await handle.signal(exitSignal);
+  await handle.result();
 });
 
 test('pendulum getGameInfoQuery', async () => {
-  const info = gameInfo({ anchorX: 1 });
-  await runPendulum(info, async (handle) => {
-    const queryResult = await handle.query(getGameInfoQuery);
-    expect(queryResult).toEqual(info);
-  });
+  const info = { ...gameInfo, anchorX: 1 };
+  const queryResult = await handle.query(getGameInfoQuery);
+  expect(queryResult).toEqual(info);
```

Comment:
```suggestion
  expect(queryResult).toEqual(initInfo);
```

The expected value of this query is not created out of thin air; it is the initial value provided on workflow creation.